File: ZAP/ZapFrontend/src/app/models/reservation/reservation.ts
export class Reservation{
    public carId : number;
    public startDate: Date;
    public endDate: Date;

    public constructor(carId: number, startDate: Date, endDate: Date){
        this.carId = carId;
        this.startDate = startDate;
        this.endDate = endDate;
    }
}

File: ZAP/ZapAPI/ZAP.DataAccess/Entities/Reservation.cs
using System;
using System.Collections.Generic;

// Code scaffolded by EF Core assumes nullable reference types (NRTs) are not used or disabled.
// If you have enabled NRTs for your project, then un-comment the following line:
// #nullable disable

namespace ZAP.DataAccess.Entities
{
    public partial class Reservation
    {
        public int ReservationId { get; set; }
        public int UserId { get; set; }
        public int CarId { get; set; }
        public int ReservationTypeId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool Incident { get; set; }

        public virtual Car Car { get; set; }
        public virtual ReservationType ReservationType { get; set; }
        public virtual User User { get; set; }
    }
}
